specreduce/tracing.py

```diff
@@ -110,3 +113,164 @@ def __post_init__(self):
                 padding = np.ma.MaskedArray(np.ones(nx - nt) * self.trace[-1], mask=True)
                 self.trace = np.ma.hstack([self.trace, padding])
         self._bound_trace()
+
+
+@dataclass
+class KosmosTrace(Trace):
+    """
+    Trace the spectrum aperture in an image.
+
+    Chops image up in bins along the dispersion (wavelength) direction,
+    fits a Gaussian within each bin to determine the trace's spatial
+    center. Finally, draws a cubic spline through the bins to up-sample
+    trace along every pixel in the dispersion direction.
+
+    (The original version of this algorithm is sourced from James
+     Davenport's `kosmos` repository.)
+
+    Parameters
+    ----------
+    image : CCDData or array-like, required
+        The image over which to run the trace. Assumes cross-dispersion
+        (spatial) direction is axis 0 and dispersion (wavelength)
+        direction is axis 1.
```

Comment:
but it has a `disp_axis` attribute.  If there is somewhere that requires this assumption, then maybe we should just raise a `ValueError` if its set to something other than 1?

Reply:
The class does assume that `disp_axis` is 1, but `disp_axis` is not an argument and our prescribed workflow doesn't include modifying it. What do you think about communicating that more implicitly by calling it `_disp_axis` versus something more explicit like you suggested?

Reply:
I think we should either do that or at least ensure its 1 and raise an error otherwise (if we expect to remove this assumption down the road, that might be nice)